### PR TITLE
Fix for csharp docs

### PR DESCRIPTION
**Pull request by madneal**
fix for the issue #4848 and #4849

### CODE REVIEW

File: docs/codeql/codeql-language-guides/analyzing-data-flow-in-csharp.rst [hunk 1]
```diff
@@ -439,7 +439,7 @@ Exercise 2
      Configuration() { this="String to System.Uri" }
      
      override predicate isSource(DataFlow::Node src) {
-       src.asExpr().hasValue()
+       src.asExpr().(Expr).hasValue()
```
Comment (hvitved) [inline docs/codeql/codeql-language-guides/analyzing-data-flow-in-csharp.rst:442]: This change is not needed, as https://github.com/github/codeql/issues/4848 will fix it instead.

**Review by hvitved — CHANGES_REQUESTED** [review]
Thanks a lot for reporting the issues and fixing them!

File: docs/codeql/codeql-language-guides/analyzing-data-flow-in-csharp.rst (hunk 1 above)
Comment (madneal) [reply docs/codeql/codeql-language-guides/analyzing-data-flow-in-csharp.rst:442]: OK. Thanks for your reply.

Comment (madneal) [reply docs/codeql/codeql-language-guides/analyzing-data-flow-in-csharp.rst]: @hvitved removed.

**Review by hvitved — APPROVED** [review]

**Pull request merged**